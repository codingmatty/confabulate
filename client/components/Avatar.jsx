import md5 from 'md5-o-matic';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const StyledAvatar = styled.img`
  border-radius: 50%;
  height: 3rem;
  width: 3rem;
`;

export default function Avatar({ email }) {
  const hash = md5(email);
  const gravatarUrl = `https://s.gravatar.com/avatar/${hash}?s=80&d=mp`;

  return <StyledAvatar src={gravatarUrl} />;
}
Avatar.propTypes = {
  email: PropTypes.string.isRequired
};
