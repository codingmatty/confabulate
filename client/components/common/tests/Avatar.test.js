import { cleanup } from 'react-testing-library';
import md5 from 'md5-o-matic';
import renderWithTheme from '../../../test-utils/render-with-theme';
import Avatar from '../Avatar';

describe('Avatar', () => {
  afterEach(cleanup);

  it('matches default snapshot', () => {
    const { container } = renderWithTheme(<Avatar email="test@email.com" />);
    expect(container).toMatchSnapshot();
  });

  it('allows image before using email', () => {
    const { container } = renderWithTheme(
      <Avatar image="image.url" email="test@email.com" />
    );
    expect(container).toMatchSnapshot();
  });

  it('includes an img with gravatar src', () => {
    const { container } = renderWithTheme(<Avatar email="test@email.com" />);
    const hash = md5('test@email.com');
    const expectedSrc = `https://s.gravatar.com/avatar/${hash}?s=96&d=mp`;
    expect(container.querySelector('img')).toHaveAttribute('src', expectedSrc);
  });

  it('does not throw when no email is provided', () => {
    expect(() => renderWithTheme(<Avatar />)).not.toThrow();
  });

  it('takes size attribute', () => {
    const { container } = renderWithTheme(
      <Avatar email="test@email.com" size={4} />
    );
    const hash = md5('test@email.com');
    const expectedSrc = `https://s.gravatar.com/avatar/${hash}?s=128&d=mp`;
    expect(container.querySelector('img')).toHaveAttribute('src', expectedSrc);
  });
});
