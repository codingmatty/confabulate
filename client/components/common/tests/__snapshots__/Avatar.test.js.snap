// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar allows image before using email 1`] = `
.c0 {
  border-radius: 50%;
  height: 3rem;
  width: 3rem;
  border: 1px solid #3B3B3B;
}

<div>
  <img
    class="c0"
    size="3"
    src="image.url"
  />
</div>
`;

exports[`Avatar matches default snapshot 1`] = `
.c0 {
  border-radius: 50%;
  height: 3rem;
  width: 3rem;
  border: 1px solid #3B3B3B;
}

<div>
  <img
    class="c0"
    size="3"
    src="https://s.gravatar.com/avatar/93942e96f5acd83e2e047ad8fe03114d?s=96&d=mp"
  />
</div>
`;
